pkg/registry/generic/rest/proxy.go

```diff
@@ -202,7 +206,40 @@ func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 
 		return tlsConn, nil
 	default:
-		return nil, fmt.Errorf("Unknown scheme: %s", h.Location.Scheme)
+		return nil, fmt.Errorf("unknown scheme: %s", h.Location.Scheme)
+	}
+}
+
+func dialURLWithTransport(httpTransport *http.Transport, location *url.URL) (net.Conn, error) {
```

Comment:
I did that originally but I think this makes it more readable, as the https case gets a bit tangled when you combine optional transport with/without optional dialer. I can change it back if you think it's worth.